### PR TITLE
Add support for default values inside functions

### LINKED ISSUE #39: Add support for default values in functions
The following file :

```
class Test
  constructor:()->
  testFn:(a,b=2)->
```

triggers error : 

```
    throw new Error('cannot traverse unknown node type: ' + node.type);
          ^
Error: cannot traverse unknown node type: DefaultParam
    at childPropertyNames (C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:70:11)
    at traverse (C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:36:47)
    at C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:27:11
    at Array.forEach (native)
    at C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:25:15
    at Array.forEach (native)
    at descend (C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:22:30)
    at traverse (C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:39:5)
    at C:\ProgramData\chocolatey\lib\nodejs.commandline.0.10.35\tools\node_modules\decaffeinate\lib\utils\traverse.js:31:9
    at Array.forEach (native)
```

### PR DESCRIPTION
**Pull request by edi9999**
For example `fn = (a=1)-> a`

Closes #39

### CODE REVIEW

**Comment by edi9999** [comment]
Closes #42 
Closes #43

**Comment by eventualbuddha** [comment]
Thanks @edi9999! I'll take a look.

File: src/utils/traverse.js [hunk 1]
```diff
@@ -102,6 +103,7 @@ const ORDER = {
   LogicalNotOp: ['expression'],
   LogicalOrOp: ['left', 'right'],
   MemberAccessOp: ['expression'],
+  MultiplyOp: ['left', 'right'],
```
Comment (eventualbuddha) [inline src/utils/traverse.js]: Is this related to default params somehow?

File: test/decaffeinate_test.js [hunk 2]
```diff
@@ -391,6 +391,91 @@ describe('automatic conversions', function() {
       check(`(a) -> a = 1`, `(function(a) { return a = 1; });`);
     });
 
+    it('does add the default value when they is any', function() {
+      check(`(a=2) -> a`, `(function(a=2) { return a; });`);
+    });
+
+    it.skip('does work with if then throw inline', function() {
```
Comment (eventualbuddha) [inline test/decaffeinate_test.js]: This, and more below, seem to be mistakenly included.

File: test/decaffeinate_test.js [hunk 3]
```diff
@@ -391,6 +391,91 @@ describe('automatic conversions', function() {
...
+          var a=b*c;
+          `);
+    });
+
+    it('should work with multiple star multiply', function() {
+      check(`a=1*2*3`, `var a=1*2*3;`);
+    });
+
+    it('should work with triple quotes', function() {
+      check(`a="""hello #{world}"""`, "var a=`hello ${world}`;");
```
Comment (eventualbuddha) [inline test/decaffeinate_test.js]: This appears to be related to the `cutLength` change above. Submit this as a different PR?

File: src/utils/traverse.js (hunk 1 above)
Comment (edi9999) [inline src/utils/traverse.js]: No, this fixes the problem with having the "multiply" operator. I didn't want to open 4 different pull requests to simplify things a bit (for me and for you), but I can if you prefer. It fixes the folowing test : https://github.com/eventualbuddha/decaffeinate/pull/40/files#diff-d2b10ceae8b94887ff3288766d636a9fR455

File: test/decaffeinate_test.js (hunk 2 above)
Comment (edi9999) [inline test/decaffeinate_test.js]: All `it.skip` are features that I found were not present (switch case doesn't work, throw inline doesn't work, for i in range doesn't work, continue statement does not work). I didn't really know how to implement them, so I thought it would be already something to include them so that it is easier to fix those bugs.

File: test/decaffeinate_test.js (hunk 3 above)
Comment (edi9999) [inline test/decaffeinate_test.js]: Would you prefer me to create pull requests for each of those added featrues ?

 * Working with `*`
 * Working with multiple `+` and multiple `*`
 * Working with triple quotes
 * Working with && and NotEq operator
 * Working with spreads
 * Working with default values
 * Adding skipped failing tests for features not yet implemented

This would probably take some time to rewrite, but If you really prefer, I can rewrite it that way

File: src/utils/traverse.js (hunk 1 above)
Comment (eventualbuddha) [inline src/utils/traverse.js]: Thanks for the explanation. I was thrown off by the difference between the PR title and the actual changes being made. No need to submit multiple PRs, but in the future it'd be nice to at least separate distinct changes into distinct commits if not PRs.

Comment (edi9999) [inline src/utils/traverse.js]: You're right, great history is also important for me. I just rewrote those commits into more logical units, and opened a new pull request : #44

**Comment by edi9999** [comment]
Close in favor of #44 (which is the same but with better ordered commits)
